client/lib/wp/sync-handler/whitelist-handler.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Module dependencies
+ */
+
+import debugFactory from 'debug';
+const debug = debugFactory( 'calypso:sync-handler:whitelist' );
+
+const whitelist = [
+	'^/me/posts$',
+	'^/me/settings',
+	'^/sites/[\\w.]+/posts'
+];
```

Comment:
Minor optimization: store the regex rather than the string:
```javascript
const whitelist = [
    /^\/me\/posts$/i,
    /^\/me\/settings/i,
    /^\/sites\/[\w.]+\/posts/i
]
```